osf/migrations/0156_pagecounter_data.py

```diff
@@ -53,8 +18,15 @@ class Migration(migrations.Migration):
         ('osf', '0155_pagecounter_schema'),
     ]
 
-    operations = [
-        migrations.RunPython(
-            separate_pagecounter_id, reverse_func
-        ),
-    ]
+    if not DEBUG_MODE:
+        operations = [
+            migrations.RunSQL([
+                FORWARD_SQL,
+                REVERSE_SQL,
+            ])
+        ]
```

Comment:
I'm pretty sure this python if statement won't actually execute, you might need to do a RunPython and execute the SQL inside of that?

Reply:
Really? I was going by [the docs](https://docs.djangoproject.com/en/1.11/ref/migration-operations/#runsql) and I think we've done it that way in other places.

Reply:
It wasn't the RunSQL bit that looked problematic by itself, but the Python `if` statement around it.

Reply:
Ah, I see. I know you can't access models or the database inside a migration (without jumping through some hoops), but I thought this would be okay. Okay, I'll poke at it and refactor as necessary, thank you.

Reply:
to be clear, I think the SQL by itself is fine, it's just there's just python code sitting that I don't think would run when the class was instantiated, so the SQL isn't going to get executed.

Reply:
Makes sense. Shouldn't be difficult to modify.

Reply:
To think this through, website/settings/defaults.py has `DEBUG_MODE = False`, and my local environment has `DEBUG_MODE = True`. The various scenarios I would expect are:

1. It gets the defaults but not the local override, in which case it would run the `not DEBUG_MODE` branch
2. It gets `None` for `DEBUG_MODE` because it's not really defined, in which case it run the `not DEBUG_MODE` branch
3. It gets a NameError because DEBUG_MODE was never defined, in which case the migration would have failed with an uncaught exception
4. It gets an ImportError for the same reasons and consequences as #3
5. It doesn't run anything in the python code, in which case the migration "runs" but doesn't do anything.

So, for all those, the only scenario that I would not expect to happen is for it to run the `else` branch on my local setup. However, when running on my local, I get `[osf.migrations.0156_pagecounter_data]  INFO: The automatic migration only runs in DEBUG_MODE. Use management command migrate_pagecount_data instead`. So I'm *pretty* sure it should work unless I'm still not understanding what you are telling me.